src/main/java/com/minecolonies/coremod/commands/AbstractCitizensCommands.java

```diff
@@ -0,0 +1,165 @@
+package com.minecolonies.coremod.commands;
+
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyManager;
+import com.minecolonies.coremod.colony.IColony;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Parent class for all citizen related commands, contains code which is the same for all commands relating citizens.
+ */
+public abstract class AbstractCitizensCommands extends AbstractSingleCommand
+{
+    private static final String NO_ARGUMENTS                    = "Please define a valid citizen and/or colony";
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param parents an array of all the parents.
+     */
+    public AbstractCitizensCommands(@NotNull final String... parents)
+    {
+        super(parents);
+    }
+
+    @NotNull
+    @Override
+    public String getCommandUsage(@NotNull final ICommandSender sender)
+    {
+        return super.getCommandUsage(sender) + "<ColonyId> <CitizenId>";
+    }
+
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        if(args.length == 0)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NO_ARGUMENTS));
+            return;
+        }
+
+        boolean firstArgumentColonyId = true;
+        int colonyId = -1;
+        if(args.length >= 2)
+        {
+            colonyId = getIthArgument(args, 0, -1);
+            if(colonyId == -1)
+            {
+                firstArgumentColonyId = false;
+            }
+        }
+
+        final Colony colony;
+        if(sender instanceof EntityPlayer && colonyId == -1)
+        {
+            IColony tempColony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), (EntityPlayer) sender);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'tempColony' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)